ramlfications/loader.py

```diff
@@ -35,7 +40,78 @@ def _yaml_include(self, loader, node):
                 return inputfile.read()
 
         with open(file_name) as inputfile:
-            return yaml.load(inputfile, self._ordered_loader)
+            parsed = yaml.load(inputfile, self._ordered_loader)
+            if file_ext == ".json":
+                parsed = self._parse_json_refs(
+                    parsed, os.path.dirname(file_name)
+                )
+            return parsed
+
+    def _lookup_json_ref(self, ref_key, base_path=None):
+        ref_source, ref_fragment = ref_key.split("#/")
```

Comment:
Hmm, so if I'm reading v4 of JSON schemas/RFCs correctly, then this doesn't work.  Please correct me if I'm wrong - there isn't a lot of docs on `$ref`s in JSON. I'm particularly looking at [this RFC draft](https://tools.ietf.org/html/draft-ietf-appsawg-json-pointer-09).

So after reading what I could find, I _think_ should be the expected behavior when parsing the value of `$ref`:

1. Full file - when the following is parsed:

  ```bash
  $ cat tests/data/examples/includes/example_with_ref.json
  {
      "second_name": "bar",
      "$ref": "example.json"
  }
  $ cat tests/data/examples/includes/example.json
  {
    "name": "foo",
    "false": true
  }
  ```

  I should get a Python dictionary of:

  ```python
  {
    "second_name": "bar",
    "name": "foo",
    "false": True
  }
 ```
  which errors out since it's expecting the `#/` (`ValueError: need more than 1 value to unpack`)

2. Partial file - when the following is parsed:

  ```bash
  $ cat tests/data/examples/includes/example_with_ref.json
  {
      "second_name": "bar",
      "$ref": "example.json#/name"
  }
  $ cat tests/data/examples/includes/example.json
  {
    "name": "foo",
    "false": true
  }
  ```

  I should get a Python dictionary of:

  ```python
  {
    "second_name": "bar",
    "name": "foo",
  }
  ```
  which actually errors out (`ValueError: need more than 1 value to unpack`)

3. Partial file - when the following is parsed:

  ```bash
  $ cat tests/data/examples/includes/example_with_ref.json
  {
      "second_name": "bar",
      "$ref": "example.json#/"
  }
  $ cat tests/data/examples/includes/example.json
  {
    "name": "foo",
    "false": true
  }

  ```

  I should get a Python dictionary of:

  ```python
  {
    "second_name": "bar"
  }
  ```

  since `#/` is referencing a property of `""` (see Section 6 of the previously referred [RFC draft](https://tools.ietf.org/html/draft-ietf-appsawg-json-pointer-09)).

5. Reference within the file:

```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",

  "definitions": {
    "address": {
      "type": "object",
      "properties": {
        "street_address": { "type": "string" },
        "city":           { "type": "string" },
        "state":          { "type": "string" }
      },
      "required": ["street_address", "city", "state"]
    }
  },
  "type": "object",
  "properties": {
    "billing_address": { "$ref": "#/definitions/address" },
    "shipping_address": {
      "allOf": [
        { "$ref": "#/definitions/address" },
        { "properties":
          { "type": { "enum": [ "residential", "business" ] } },
          "required": ["type"]
        }
      ]
    }
  }
}
```
so that `shipping_address` includes the properties of `definitions/address`.